src/common/types/typedef.ts

```diff
@@ -272,6 +280,56 @@ const addBuiltinTypes = (definitions: TypeDefinitions) => {
         ])
     );
 
+    /**
+     * ```
+     * alias UpscaleChannels {
+     *   imageChannels: int(1..inf),
+     *   inputChannels: int(1..inf),
+     *   outputChannels: int(1..inf),
+     * } = match imageChannels {
+     *   1 => 1,
+     *   4 => match inputChannels { 3 => add(outputChannels, 1), _ => outputChannels },
+     *   _ => outputChannels
+     * }
+     * ```
+     */
+    definitions.add(
+        new AliasDefinition(
```

Comment:
I'm honestly considering adding a parser for type expressions and statements. I mean, look at this!

Reply:
That's probably a good idea. The one you did for text append was pretty gnarly as well

Reply:
Yeah, I think I'll throw some ANTLR4 stuff together. Or do you know any other parser generators for JS/TS?

Reply:
I do not.